File: sdks/java/PrakashSDK.java
package com.prakash.sdk;

import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.time.Duration;
import java.util.HashMap;
import java.util.Map;
import com.fasterxml.jackson.databind.ObjectMapper;

/**
 * Prakash Billing System - Java SDK
 * For Central Dashboard Integration
 */
public class PrakashSDK {
    private final String tenantId;
    private final String apiKey;
    private final String endpoint;
    private final HttpClient httpClient;
    private final ObjectMapper objectMapper;

    public PrakashSDK(String tenantId, String apiKey, String endpoint) {
        if (tenantId == null || tenantId.isEmpty()) {
            throw new IllegalArgumentException("Tenant ID is required.");
        }
        this.tenantId = tenantId;
        this.apiKey = apiKey;
        this.endpoint = (endpoint != null && !endpoint.isEmpty()) 
            ? endpoint 
            : "https://dashboard.prakash.com/api/central/ingest";
        
        this.httpClient = HttpClient.newBuilder()
            .version(HttpClient.Version.HTTP_2)
            .connectTimeout(Duration.ofSeconds(10))
            .build();
        this.objectMapper = new ObjectMapper();
    }

    public void logSubmission(String documentId, String status, Map<String, Object> details) {
        Map<String, Object> payload = new HashMap<>();
        payload.put("tenantId", this.tenantId);
        payload.put("eventType", "submission");
        payload.put("documentId", documentId);
        payload.put("status", status);
        payload.put("details", details);

        sendEvent(payload);
    }

    public void logError(String context, Exception error) {
        Map<String, Object> details = new HashMap<>();
        details.put("context", context);
        details.put("errorMessage", error.getMessage());
        
        // Simplified stack trace
        if (error.getStackTrace().length > 0) {
            details.put("stack", error.getStackTrace()[0].toString());
        }

        Map<String, Object> payload = new HashMap<>();
        payload.put("tenantId", this.tenantId);
        payload.put("eventType", "error");
        payload.put("status", "failure");
        payload.put("details", details);

        sendEvent(payload);
    }

    private void sendEvent(Map<String, Object> payload) {
        try {
            String jsonPayload = objectMapper.writeValueAsString(payload);
            
            HttpRequest request = HttpRequest.newBuilder()
                .uri(URI.create(this.endpoint))
                .timeout(Duration.ofMinutes(1))
                .header("Content-Type", "application/json")
                .header("Authorization", "Bearer " + (this.apiKey != null ? this.apiKey : ""))
                .header("X-Tenant-ID", this.tenantId)
                .POST(HttpRequest.BodyPublishers.ofString(jsonPayload))
                .build();

            httpClient.sendAsync(request, HttpResponse.BodyHandlers.ofString())
                .thenApply(HttpResponse::body)
                .thenAccept(System.out::println)
                .exceptionally(e -> {
                    System.err.println("Failed to log event to Prakash Dashboard: " + e.getMessage());
                    return null;
                });

        } catch (Exception e) {
            System.err.println("Error constructing request: " + e.getMessage());
        }
    }
    
    // Usage Example
    public static void main(String[] args) {
        PrakashSDK sdk = new PrakashSDK("test-tenant-123", "secret-key", null);
        Map<String, Object> details = new HashMap<>();
        details.put("amount", 1000);
        sdk.logSubmission("INV-2024-001", "success", details);
    }
}

File: sdks/python/prakash_sdk.py
import os
import requests
import json
from typing import Optional, Dict, Any, Union

class PrakashSDK:
    def __init__(self, tenant_id: str = None, api_key: str = None, endpoint: str = None):
        """
        Initialize the Prakash SDK client for Python.
        
        :param tenant_id: The unique identifier for the tenant (optional, defaults to env PRAKASH_TENANT_ID)
        :param api_key: The API key for authentication (optional, defaults to env PRAKASH_API_KEY)
        :param endpoint: The central dashboard API endpoint (optional, defaults to production URL)
        """
        self.tenant_id = tenant_id or os.getenv('PRAKASH_TENANT_ID')
        self.api_key = api_key or os.getenv('PRAKASH_API_KEY')
        self.endpoint = endpoint or 'https://dashboard.prakash.com/api/central/ingest'
        
        if not self.tenant_id:
            raise ValueError("Tenant ID is required. Pass it in init or set PRAKASH_TENANT_ID environment variable.")

    def log_submission(self, document_id: str, status: str, details: Optional[Dict[str, Any]] = None):
        """
        Log a document submission event.
        
        :param document_id: The unique ID of the document (e.g. invoice number)
        :param status: The status of the submission ('success' or 'failure')
        :param details: Additional details about the submission
        """
        payload = {
            'tenantId': self.tenant_id,
            'eventType': 'submission',
            'documentId': document_id,
            'status': status,
            'details': details or {}
        }
        return self._send_event(payload)

    def log_error(self, context: str, error: Union[Exception, str]):
        """
        Log an error event.
        
        :param context: Where the error occurred
        :param error: The error object or message
        """
        error_message = str(error)
        stack_trace = getattr(error, '__traceback__', None) if isinstance(error, Exception) else None
        
        payload = {
            'tenantId': self.tenant_id,
            'eventType': 'error',
            'status': 'failure',
            'details': {
                'context': context,
                'errorMessage': error_message,
                'stack': str(stack_trace) if stack_trace else None
            }
        }
        return self._send_event(payload)

    def _send_event(self, payload: Dict[str, Any]):
        headers = {
            'Content-Type': 'application/json',
            'Authorization': f'Bearer {self.api_key or ""}',
            'X-Tenant-ID': self.tenant_id
        }
        
        try:
            response = requests.post(self.endpoint, json=payload, headers=headers, timeout=10)
            response.raise_for_status()
            return response.json()
        except requests.exceptions.RequestException as e:
            print(f"Failed to log event to Prakash Dashboard: {e}")
            return None

# Usage Example:
if __name__ == "__main__":
    sdk = PrakashSDK(tenant_id="test-tenant-123", api_key="secret-key")
    sdk.log_submission("INV-2024-001", "success", {"amount": 1000})
